app/models/user.rb

```diff
@@ -2,4 +2,8 @@ class User < ApplicationRecord
   has_many :comments
   has_many :likes
   has_many :posts
+
+  def recent_post
+    posts.order(created_at: :desc).limit(3)
+  end
```

Comment:
- [ ] According to the ERD diagram, the foreign_key on the `comments`, `likes`, and `posts` tables are supposed to be `author_id` instead of `user_id`, therefore in order to comply with requirements, you need to rename the columns on the aforementioned tables.
![image](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png?raw=true)
- [ ] Doing this would mean that you need to modify the associations like this:
```ruby
has_many :posts, foreign_key: 'author_id'
has_many :comments, foreign_key: 'author_id'
has_many :likes, foreign_key: 'author_id'
```

Reply:
@codepantha Hi, I was advised previously to change it to `t.references :author, null: false, foreign_key: { to_table: 'users' }`, but maybe I misunderstood. You can see this request here: https://github.com/sebalp100/blog-app/pull/1#discussion_r1130300689. Should I remove this and replace it?

Reply:
If you look at your `schema.rb` file, you will see that you have `user_id` column instead of `author_id` on the posts, comments and likes tables. You need to create a migration file to rename the affected column.

Please use this [resource](https://www.dailysmarty.com/posts/how-to-rename-a-column-in-ruby-on-rails) in case you need help